internal/pkg/externalplugins/cherrypicker/cherrypicker_test.go

```diff
@@ -401,16 +401,28 @@ func testCherryPickIC(clients localgit.Clients, t *testing.T) {
 	if err := s.handleIssueComment(logrus.NewEntry(logrus.StandardLogger()), ic); err != nil {
 		t.Fatalf("unexpected error: %v", err)
 	}
+
 	if len(ghc.prs) != len(expectedBranches) {
 		t.Fatalf("Expected %d PRs, got %d", len(expectedBranches), len(ghc.prs))
 	}
 
-	for i, expectedBranch := range expectedBranches {
-		got := prToString(ghc.prs[i])
-		expected := expectedFn(expectedBranch)
-		if got != expected {
-			t.Errorf("Expected (%d):\n%s\nGot (%d):\n%+v\n", len(expected), expected, len(got), got)
+	expectedPrs := make(map[string]string)
+	for _, branch := range expectedBranches {
+		expectedPrs[expectedFn(branch)] = branch
+	}
+
+	seenBranches := make(map[string]struct{})
+	for _, p := range ghc.prs {
+		pr := prToString(p)
+		branch, present := expectedPrs[pr]
+		if !present {
+			t.Errorf("Unexpected PR:\n%s\nExpected to target one of the following branches: %v\n",
+				pr, expectedBranches)
 		}
+		seenBranches[branch] = struct{}{}
```

Comment:
Is it possible to use the `sets.NewString()` directly, or change `struct{}` to bool, it seems that we don’t need to store a struct.

Reply:
Fixed.

Reply:
Fxied.